extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +84,48 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+    function loadHQImage_() {
+      // We only need to handle error once to load another source.
+      imgPlaceholder.removeEventListener('error', loadHQImage_);
+      imgPlaceholder.setAttribute('src', 'https://i.ytimg.com/vi/' +
+          videoid + '/hqdefault.jpg');
+    }
+
+    function maybeLoadHQImage_() {
+      // We only need to handle error once to load another source.
+      imgPlaceholder.removeEventListener('load', maybeLoadHQImage_);
+      // A pretty ugly hack since onerror won't fire on YouTube image 404.
+      // This might be due to the fact that YouTube returns data to the request
+      // even when the status is 404.
+      if ((this.naturalWidth === 120 && this.naturalHeight === 90) ||
```

Comment:
just an fyi (no need to change unless somebody else chimes in), we arent strict about it but the team usually uses `==` and only use `===` when necessary. granted we do this because our code base is supposed to be "typed" which our compiler currently doesnt enforce.